test/system/HashMapTest.ooc

```diff
@@ -177,6 +182,33 @@ HashMapTest: class extends Fixture {
 
 			hashmap free()
 		})
+		this add("to and from file", func {
+			hashmap := HashMap<String, String> new()
+			hashmap put("pi", "3.14159")
+			hashmap put("e", "2.71828")
+			hashmap put("sqrt(2)", "1.41459")
+			hashmap put("primes", "2,3,5,7,11,13,17")
+			hashmap writeToFile("test/system/output/mathmap.txt")
+			hashmap free()
+
+			tolerance := 0.00001f
+			input := HashMap readFromFile(t"test/system/output/mathmap.txt")
+			expect(input, is notNull)
+			expect(input count, is equal to(4))
+			(piString, eString, sqrtTwoString, primeString) := (input get("pi"), input get("e"), input get("sqrt(2)"), input get("primes"))
+			(pi, e, sqrtTwo) := (piString, eString, sqrtTwoString) toFloat()
+			expect(pi, is equal to(3.14159f) within(tolerance))
+			expect(e, is equal to(2.71828f) within(tolerance))
+			expect(sqrtTwo, is equal to(1.41459f) within(tolerance))
+			primes := primeString split(',')
+			expect(primes count, is equal to(7))
+			primes free()
+
+			// Since keys/values are not string literals we need to manually free
+			input each(freeFunc)
```

Comment:
I see your point, but...

> So this will be needed everytime

Yes, but it's not very different from other pieces of ooc: If you throw classes in a generic something, you are also responsible for dealing with their destruction. Having a function which iterates over all entries and frees the key+value pair I thought was actually pretty neat. :)

> is there some other way

No. :( I mean, we could have a method in the map like `freeAllKeyValuePairs()` which would contain a huge switch statement and try to free "all" types of keys and values... but that's not too practical. We *could* have a specialized one for just the <String, String> type, but that may be a bit too much specialization. What do you say?